packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useState } from 'react';
+
+import { useAppContext } from '../../context/app';
+import { useUserContext } from '../../context/user';
+import { useCartContext } from '../../context/cart';
+
+export const useCheckoutPage = () => {
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ isSignedIn }] = useUserContext();
+    const [{ isEmpty }] = useCartContext();
+
+    /**
+     * Using local state to maintain these booleans. Can be
+     * moved to checkout context in the future if needed.
+     */
+    const [shippingInformationDone, setShippingInformationDone] = useState(
```

Comment:
One way to tell if shipping information/method/etc are done is to just make a query for the data and see if anything returns. In the case of shipping information we will probably want to do something like query for shipping address as the estimator on cart does but then wipe the cache of any of that fake data.

I'd definitely prefer that we allow the apollo cache to control the "renderable" state of things. For example:

* checkout page would query for shipping information (along with other info such as shipping method, pre-added gift card/coupons/etc) which dictate what step to display.
* shipping address component would query for any provided cart shipping address which, if found, would tell it to render in the "card" UI rather than the form UI.

As it is, on a page refresh you lose local state which means you would have to start over each time when in reality part of your data would have already been saved to gql.

Reply:
You are absolutely right about losing data on refresh which is not what we want. We can use apollo client local queries for that. Also, these boolean flags here are used to track progress, not data presence. Even if you have shipping method picked in cart page, and that data is available at the time of checkout, you still need them to click on proceed to payment information but use the data from prev step or from the account (shipping address for instance). This is what I understood after I spoke with @soumya-ashok a couple of days ago. That is the reason I have those progress indicator flags. They have to be moved to a persistent store, but for now, since this is just the scaffolding PR, I have left them in the hook.

Reply:
>  Also, these boolean flags here are used to track progress, not data presence.

Right, I'm just implying that data presence itself is an indicator of the progress a user has made so there shouldn't be a need to re-submit it, which is what those "proceed to X" buttons are for. 

Assume a user has entered (and saved to the server) a bunch of information on their cart and they're on some final step like payment method. They've clicked "proceed to X step" a bunch of time at that point. If they refresh the page, for whatever reason, they will have to click through all those steps again, likely resubmitting redundant data that the server is already aware of.

Reply:
You are right. Let's see what Soumya and Scott have in mind.